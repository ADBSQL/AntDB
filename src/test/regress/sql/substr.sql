set grammar to oracle;
set datestyle='ISO,YMD';
SELECT SUBSTR('ABCDEFG',3,4) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',0,4) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',-2,4) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',-7,10) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',-8,10) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',1,0) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',1,-1) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',1,10) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',100000000,1) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',2,10000000) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',2,10000000) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG','1','2') sub FROM DUAL;
SELECT SUBSTR(1234,'1','2') sub FROM DUAL;
SELECT SUBSTR(1234.4,1.1,2.0) sub FROM DUAL;
SELECT SUBSTR(to_char('abcd'),1,2) sub FROM DUAL;
SELECT SUBSTR(exp(2),1,6) sub FROM DUAL;
SELECT SUBSTR('abcde',tanh(1),6) sub FROM DUAL;
SELECT SUBSTR(to_date('2015-06-06 20:50:30','yyyy-mm-dd hh24:mi:ss'),1,10) sub FROM DUAL;
SELECT SUBSTR(to_timestamp('2015-06-06 20:50:30','yyyy-mm-dd hh24:mi:ss'),1,10) sub FROM DUAL;
SELECT SUBSTR('ABCDEFG',2) sub FROM DUAL;
SELECT SUBSTR('',2) sub FROM DUAL;
SELECT SUBSTR(null,2) sub FROM DUAL;
SELECT SUBSTR('abc','') sub FROM DUAL;
