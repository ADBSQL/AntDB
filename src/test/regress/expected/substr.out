set grammar to oracle;
set datestyle='ISO,YMD';
SELECT SUBSTR('ABCDEFG',3,4) sub FROM DUAL;
 SUB  
------
 CDEF
(1 row)

SELECT SUBSTR('ABCDEFG',0,4) sub FROM DUAL;
 SUB  
------
 ABCD
(1 row)

SELECT SUBSTR('ABCDEFG',-2,4) sub FROM DUAL;
 SUB 
-----
 FG
(1 row)

SELECT SUBSTR('ABCDEFG',-7,10) sub FROM DUAL;
   SUB   
---------
 ABCDEFG
(1 row)

SELECT SUBSTR('ABCDEFG',-8,10) sub FROM DUAL;
 SUB 
-----
 
(1 row)

SELECT SUBSTR('ABCDEFG',1,0) sub FROM DUAL;
 SUB 
-----
 
(1 row)

SELECT SUBSTR('ABCDEFG',1,-1) sub FROM DUAL;
 SUB 
-----
 
(1 row)

SELECT SUBSTR('ABCDEFG',1,10) sub FROM DUAL;
   SUB   
---------
 ABCDEFG
(1 row)

SELECT SUBSTR('ABCDEFG',100000000,1) sub FROM DUAL;
 SUB 
-----
 
(1 row)

SELECT SUBSTR('ABCDEFG',2,10000000) sub FROM DUAL;
  SUB   
--------
 BCDEFG
(1 row)

SELECT SUBSTR('ABCDEFG',2,10000000) sub FROM DUAL;
  SUB   
--------
 BCDEFG
(1 row)

SELECT SUBSTR('ABCDEFG','1','2') sub FROM DUAL;
 SUB 
-----
 AB
(1 row)

SELECT SUBSTR(1234,'1','2') sub FROM DUAL;
 SUB 
-----
 12
(1 row)

SELECT SUBSTR(1234.4,1.1,2.0) sub FROM DUAL;
 SUB 
-----
 12
(1 row)

SELECT SUBSTR(to_char('abcd'),1,2) sub FROM DUAL;
 SUB 
-----
 ab
(1 row)

SELECT SUBSTR(exp(2),1,6) sub FROM DUAL;
  SUB   
--------
 7.3890
(1 row)

SELECT SUBSTR('abcde',tanh(1),6) sub FROM DUAL;
  SUB  
-------
 abcde
(1 row)

SELECT SUBSTR(to_date('2015-06-06 20:50:30','yyyy-mm-dd hh24:mi:ss'),1,10) sub FROM DUAL;
    SUB     
------------
 2015-06-06
(1 row)

SELECT SUBSTR(to_timestamp('2015-06-06 20:50:30','yyyy-mm-dd hh24:mi:ss'),1,10) sub FROM DUAL;
    SUB     
------------
 2015-06-06
(1 row)

SELECT SUBSTR('ABCDEFG',2) sub FROM DUAL;
  SUB   
--------
 BCDEFG
(1 row)

SELECT SUBSTR('',2) sub FROM DUAL;
 SUB 
-----
 
(1 row)

SELECT SUBSTR(null,2) sub FROM DUAL;
 SUB 
-----
 
(1 row)

SELECT SUBSTR('abc','') sub FROM DUAL;
 SUB 
-----
 
(1 row)

